JavaScript/JavaScriptSDK/ajax/fetch.ts

```diff
@@ -0,0 +1,229 @@
+﻿/// <reference path="../Logging.ts" />
+/// <reference path="../Util.ts" />
+/// <reference path="./ajaxUtils.ts" />
+/// <reference path="./ajaxRecord.ts" />
+/// <reference path="../RequestResponseHeaders.ts" />
+/// <reference path="../Telemetry/RemoteDependencyData.ts" />
+/// <reference path="../AppInsights.ts" />
+
+module Microsoft.ApplicationInsights {
+    "use strict";
+
+    export class FetchMonitor {
+        private appInsights: AppInsights;
+        private initialized: boolean;
+        public static instrumentedByAppInsightsName = "InstrumentedByAppInsights";
+        private currentWindowHost;
+
+        constructor(appInsights: Microsoft.ApplicationInsights.AppInsights) {
+            this.currentWindowHost = window.location.host && window.location.host.toLowerCase();
+            this.appInsights = appInsights;
+            this.initialized = false;
+            this.Init();
+        }
+
+        private Init(): void {
+            if (this.supportsMonitoring()) {
+                this.instrumentFetch();
+                this.initialized = true;
+            }
+        }
+
+        public static DisabledPropertyName: string = "Microsoft_ApplicationInsights_BypassFetchInstrumentation";
```

Comment:
Unused, did you want to add if fetch request have this property set so that those are not monitored?